pkg/authn/keychain.go

```diff
@@ -178,3 +179,62 @@ func (w wrapper) Resolve(r Resource) (Authenticator, error) {
 	}
 	return FromConfig(AuthConfig{Username: u, Password: p}), nil
 }
+
+type refreshingKeychain struct {
```

Comment:
wdyt about exporting this as `authn.RefreshingKeychain(inner authn.Keychain, d time.Duration)`

Reply:
Sure, done.